CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -52,6 +52,108 @@ class CRM_Contribute_Form_Contribution_Confirm extends CRM_Contribute_Form_Contr
   public $_contributionID;
 
   /**
+   * @param $form
+   * @param $params
+   * @param $contributionParams
+   * @param $pledgeID
+   * @param $contribution
+   * @param $isEmailReceipt
+   * @return mixed
+   */
+  public static function handlePledge(&$form, $params, $contributionParams, $pledgeID, $contribution, $isEmailReceipt) {
```

Comment:
This is all just an extraction - moving a chunk of code out of the main function